env/git.go

```diff
@@ -112,26 +112,21 @@ func GitSHA(path string) (string, error) {
 }
 
 var GitBlob = func(path string, commit *object.Commit) (string, error) {
-	if commit == nil {
-		blob, err := fallbackBlob(path)
+	if commit != nil {
+		if file, err := commit.File(path); err == nil {
```

Comment:
yep, exactly this would fallback only if we don't know the commit or if we don't know the file